zip-bolt-support.rst

```diff
@@ -0,0 +1,408 @@
+
+* ``version``: 2
+* ``groupid``: specify group id
+* ``locktime``: 0
+* ``txin`` count: 1
+   * ``txin[0]`` outpoint: ``txid`` and ``output_index``
+   * ``txin[0]`` sequence: 0xFFFFFFFF
+   * ``txin[0]`` script bytes: 0
+   * ``txin[0]`` script sig: ``0 <cust-sig> <merch-sig>``
+* ``nShieldedOutput``: 1
+* ``vShieldedOutput[0]``:
+   - ``cv``: commitment for the output note
+   - ``cmu``: ...
+   - ``ephemeralKey``:ephemeral public key
+   - ``encCiphertext``: encrypted output note (part 1)
+   - ``outCiphertext``: encrypted output note (part 2)
+   - ``zkproof``: zero-knowledge proof for the note
+
+The merchant can initiate closing by posting the initial commitment transaction (in Section 2.3) from establishing the channel that pays the merchant the full balance of the channel with a time lock that allows for customer dispute. The merchant can then post a separate standard transaction that moves those funds to a shielded address.
+
+3. Custom Shielded Tx: Using Z-addresses and Scriptless
+-------------
+We assume the following features are present:
+
+(a) ``lock_time`` - for absolute lock time
+(b) A way to enforce relative lock time
+(c) 2-of-2 multi-sig shielded address support
+(d) All inputs/outputs are specified from/to a shielded address
+(e) A method to encumber the outputs of a shielded transaction
+(f) An extension to the transaction format to include BOLT (e.g., like ``vBoltDescription``)
+(g) Extend the ``SIGHASH`` flags to cover the extended field
+
+The goal here is to perform all the same validation steps for channel opening/closing without relying on the scripting system, as well as allowing for relative timelocks (the equivalent of ``OP_CSV``). In order to support multihop payments, we need absolute timelocks as well (the equivalent of ``OP_CLTV``). We also want to ensure that transactions are non-malleable in order to allow for unconfirmed dependency transaction chains.
+
+**Limitations/Notes**: With extensions to shielded transaction format, it may be evident whenever parties are establishing private payment channels. We appreciate feedback on the feasibility of what is proposed for each aspect of the Bolt protocol.
+
+3.1 Channel Opening
+-------------
+The customer creates a funding transaction that spends ZEC from a shielded address to a 2-of-2 multi-sig shielded address. Here is the flow (1) creating a multisig shielded address specifying both parties keys and (2) generating channel tokens.
+
+3.2 Funding Transaction
+-------------
+The funding transaction is by default funded by only one participant, the customer. It could also be funded by the merchant. 
+
+This transaction has 2 shielded inputs (but can be up to some N) and 1 output to a 2-of-2 shielded address with the merchant public key. If an ``vBoltDescription`` field is added, then we could use it to store the channel parameters and the channel token for opening the channel.
```

Comment:
This paragraph appears to assume an example transaction that isn't specified here. Maybe just remove the first sentence.